### PR TITLE
Clean up permissions for student reg

### LINKED ISSUE #2059: Student/Classes and Student/Classes/OneClass permissions need help text
And possibly also need cleanup of what they do, since most programs setting one will need to set the other. OneClass seems to be intended for a use case that I'm not sure anyone is actually using anymore.

### PR DESCRIPTION
**Pull request by mgersh**
Includes: delete Student/Classes/OneClass and switch all uses to
Student/Classes, add back Student/Removal which seems to have been
accidentally deleted in main but still exists on Stanford's branch,
and delete scrmi.signup_verb. Fixes #2059 and #1882.

### CODE REVIEW

File: esp/esp/program/modules/handlers/studentclassregmodule.py [hunk 1]
```diff
@@ -330,24 +330,8 @@ def addclass_logic(self, request, tl, one, two, module, extra, prog):
         """ Pre-register the student for the class section in POST['section_id'].
             Return True if there are no errors.
         """
-        reg_perm = 'Student/Classes'
         scrmi = self.program.studentclassregmoduleinfo
 
-        if 'prereg_verb' in request.POST:
-            proposed_verb = "V/Flags/Registration/%s" % request.POST['prereg_verb']
-            if scrmi.use_priority:
-                available_verbs = ["%s/%d" % (scrmi.signup_verb.get_uri(), x) for x in xrange(1, scrmi.priority_limit+1)]
```
Comment (benjaminjkraft) [inline esp/esp/program/modules/handlers/studentclassregmodule.py]: Whoa, I was going to ask if this was ever used, and apparently `get_uri()` doesn't even exist anymore, so I guess not!

File: esp/esp/program/modules/migrations/0013_auto_20160709_2038.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('modules', '0012_auto_20160212_1903'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
```
Comment (benjaminjkraft) [inline esp/esp/program/modules/migrations/0013_auto_20160709_2038.py]: Seems like this part was probably included by accident?

File: esp/esp/program/modules/module_ext.py [hunk 3]
```diff
@@ -137,7 +124,8 @@ def module(self):
         return self.program.getModule('StudentClassRegModule')
 
     def reg_verbs(self):
-        verb_list = [self.signup_verb]
+        verb_list = [RegistrationType.get_map(include=['Enrolled'],
```
Comment (benjaminjkraft) [inline esp/esp/program/modules/module_ext.py]: It looks like if you want you can just inline this to `RegistrationType.get_or_create(...)`.

File: esp/esp/program/tests.py [hunk 4]
```diff
@@ -515,8 +515,10 @@ def studentreg(self):
 
         sr = StudentRegistration.objects.all()[0]
 
+        enrolled = RegistrationType.get_map(include=['Enrolled'],
```
Comment (benjaminjkraft) [inline esp/esp/program/tests.py]: Likewise here.

File: esp/esp/users/models/__init__.py [hunk 5]
```diff
@@ -2209,14 +2209,14 @@ class Permission(ExpirableModel):
...
-            ("Student/Classes", "Classes"),
-            ("Student/Classes/OneClass", "Classes/OneClass"),
+            ("Student/Classes", "Register for classes"),
             ("Student/Classes/Lottery", "Enter the lottery"),
             ("Student/Classes/Lottery/View", "View lottery results"),
             ("Student/ExtraCosts", "Extra costs page"),
             ("Student/MainPage", "Registration mainpage"),
             ("Student/Confirm", "Confirm registration"),
             ("Student/Cancel", "Cancel registration"),
+            ("Student/Removal", "Remove class registrations after registration closes"),
```
Comment (benjaminjkraft) [inline esp/esp/users/models/__init__.py]: I think this is also extraneous?

**Comment by benjaminjkraft** [comment]
:raised_hands: just a few minor things

File: esp/esp/program/modules/module_ext.py (hunk 3 above)
Comment (mgersh) [inline esp/esp/program/modules/module_ext.py]: Done.

File: esp/esp/program/tests.py (hunk 4 above)
Comment (mgersh) [inline esp/esp/program/tests.py]: Done.

File: esp/esp/program/modules/migrations/0013_auto_20160709_2038.py (hunk 2 above)
Comment (mgersh) [inline esp/esp/program/modules/migrations/0013_auto_20160709_2038.py]: If by accident you mean "we accidentally left it out of #2049", then yes. I left it here on purpose though :stuck_out_tongue:

Comment (benjaminjkraft) [inline esp/esp/program/modules/migrations/0013_auto_20160709_2038.py]: Ah, got it.

**Pull request merged**